src/serverd.cpp

```diff
@@ -0,0 +1,184 @@
+    server_addr->sun_family = AF_UNIX;\
+    // TODO: add check for sock_path length
+    strcpy(static_cast<char*>(server_addr->sun_path), sock_path);
+
+    memset(client_addr, 0, addr_len);
+
+    if (std::remove(sock_path) == ErrorCode && errno != ENOENT) {
+        throw std::string_view{"Can't delete socket file"};
+    }
+}
+
+auto create_socket(int* server_sock) -> void
+{
+    *server_sock = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (*server_sock == ErrorCode) {
+        throw std::string_view{std::format("Socket error: %d", errno)};
+    }
+}
+
+auto bind_socket(
+    int* server_sock,
+    struct sockaddr_un* server_addr,
+    socklen_t addr_len
+) -> void {
+    if (bind(
+        *server_sock,
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+        reinterpret_cast<struct sockaddr*>(server_addr),
+        addr_len
+    ) == ErrorCode) {
+        throw std::string_view{std::format("Bind socket error: %d", errno)};
+    }
+}
+
+auto listen_socket(int* server_sock, const int backlog_size) -> void
+{
+    if (listen(*server_sock, backlog_size) == ErrorCode) {
+        throw std::string_view{std::format("Listen socket error: %d", errno)};
+    }
+}
+
+auto accept_client(
+    int* server_sock,
+    int* client_sock,
+    struct sockaddr_un* client_addr,
+    socklen_t* addr_len
+) -> void {
+    *client_sock = accept(
+        *server_sock,
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+        reinterpret_cast<struct sockaddr*>(client_addr),
+        addr_len
+    );
+    if (*client_sock == ErrorCode) {
+        throw std::string_view{"Can't accept client"};
+    }
+}
+
+auto log_peer_name(
+    int* client_sock,
+    struct sockaddr_un* client_addr,
+    socklen_t* addr_len
+) -> void {
+    if (getpeername(
+        *client_sock,
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+        reinterpret_cast<struct sockaddr*>(client_addr),
+        addr_len
+    ) == ErrorCode) {
+        // TODO: handle error
+    }
+    // TODO: return "client_sockaddr.sun_path"
+}
+
+auto recieve_method(int* client_sock) -> Query_method
+{
+    Query_method query_method{Query_method::EnumEnd};
+    if (recv(
+        *client_sock,
+        &query_method,
+        BytesForSize,
+        0
+    ) == ErrorCode) {
+        throw std::string_view{"Can't get request method"};
+    }
+    return query_method;
+}
+
+auto handle_request(int* client_sock, Query_method query_method) -> void
+{
+    if (
+        query_method < Query_method::EnumStart
+        || query_method >= Query_method::EnumEnd
+    ) {
+        throw std::string_view{"Unknown query method"};
+    }
+    // TODO: call matching method
+
+    if (send(
+        *client_sock,
+        &query_method,
+        BytesForSize,
+        0
+    ) == ErrorCode) {
+        throw std::string_view{"Can't send answer"};
+    }
+}
+
+auto serve(const char* sock_path, const int backlog_size) -> void
```

Comment:
Accepting values as `const` is not wrong, but certainly unconventional. In a not so often case when you really try to change value of in-parameter without using it later, compiler should catch it, and issue a warning.